test/sqllogictest/transform/join_fusion.slt

```diff
@@ -155,26 +155,27 @@ Explained Query:
     cte l1 =
       Reduce aggregates=[min(#0)] // { arity: 1 }
         Project (#1) // { arity: 1 }
-          Filter (#1 <= 195) AND (#1 >= 38) AND (1997-08-25 00:00:00 = date_to_timestamp(#4)) // { arity: 6 }
-            Join on=(#0 = #3 AND #1 = #5) type=differential // { arity: 6 }
-              implementation
-                %2[#0] » %1:orders[#0]KAeiiiif » %0:lineitem[#0]KAeiiiif
-              ArrangeBy keys=[[#0]] // { arity: 1 }
-                Project (#4) // { arity: 1 }
-                  Filter (#6 = 1997-01-25) // { arity: 8 }
-                    Get materialize.public.lineitem // { arity: 8 }
-              ArrangeBy keys=[[#0]] // { arity: 4 }
-                Get materialize.public.orders // { arity: 4 }
-              ArrangeBy keys=[[#0]] // { arity: 1 }
-                Union // { arity: 1 }
-                  Project (#0) // { arity: 1 }
-                    Get l0 // { arity: 9 }
-                  Map (error("more than one record produced in subquery")) // { arity: 1 }
-                    Project () // { arity: 0 }
-                      Filter error("more than one record produced in subquery") AND (#0 > 1) // { arity: 1 }
-                        Reduce aggregates=[count(true)] // { arity: 1 }
-                          Project () // { arity: 0 }
-                            Get l0 // { arity: 9 }
+          Join on=(#0 = #2 AND #1 = #3) type=differential // { arity: 4 }
+            implementation
+              %1:orders[#1]KAeiif » %0:lineitem[#0]KAeiif » %2[#0]KAeiif
+            ArrangeBy keys=[[#0]] // { arity: 1 }
+              Project (#4) // { arity: 1 }
+                Filter (#6 = 1997-01-25) // { arity: 8 }
+                  Get materialize.public.lineitem // { arity: 8 }
+            ArrangeBy keys=[[#1]] // { arity: 2 }
+              Project (#0, #2) // { arity: 2 }
+                Filter (#0 <= 195) AND (#0 >= 38) AND (1997-08-25 00:00:00 = date_to_timestamp(#3)) // { arity: 4 }
```

Comment:
This looks like a possible regression, where we've gone from a primary key join (on `o_orderkey`) to a join on `o_orderdate`. I can't really read the plans to be honest, but that is what seems to have happened.

Reply:
Btw. [if our key detection would realize that l0 (a scalar subquery) also has a unique key](https://github.com/MaterializeInc/materialize/issues/16501), then that would also get us the right order here.